src/GraphQL/Server.hs

```diff
@@ -76,11 +73,34 @@ infixr 8 :<>
 data a :<|> b = a :<|> b
 infixr 8 :<|>
 
+
+-- Result collects errors and values at the same time unless a handler
+-- tells us to bail out in which case we stop the processing
+-- immediately.
+data Result a = Result [ResolverError] a deriving (Show, Functor)
+
+-- Aggregating results keeps all errors and creates a ValueList
+-- containing the individual values.
+aggregateResults :: [Result GValue.Value] -> Result GValue.Value
+aggregateResults r =
+  let (errs, valueList) = foldl' (\(eb, vb) (Result ea va) -> ((eb <> ea), va:vb)) ([], []) r
+  in Result errs (GValue.toValue valueList)
+
+
+instance Applicative Result where
+  pure v = Result [] v
+  -- TODO it's not obvious to me that this is the right function...
+  (Result e f) <*> (Result ea ev) = Result (e <> ea) (f ev)
```

Comment:
Looks right to me. We always apply the function (whence 'applicative' gets its terrible name) and our context always appends errors. 

If in doubt, write some property tests for the applicative laws.

As a style thing, I'd say `e1` and `e2`, and change `ev` to `x`, but that's mostly personal preference.